src/volunteer-form/index.tsx

```diff
@@ -35,6 +37,7 @@ type VolunteerFormState = {
   readonly submittingTutor: boolean;
   readonly submittedMentor: boolean;
   readonly submittedTutor: boolean;
+  error: ApiCallResult | undefined;
```

Comment:
How about we don't add any more props and just use `await-to-js` to catch the errors thrown by `this.context.signup()`?

Like so:
```typescript
import { ApiError } from '@tutorbook/model';
import to from 'await-to-js';
...
const [error] = await to<void, ApiError>(this.context.signup(this.context.user));
if (error) this.setState({ error }); // Show the error dialog.
// Otherwise, just continue as normal and show the checkmark.
```